index.js

```diff
@@ -361,22 +346,20 @@ module.exports = class Hyperdrive extends ReadyResource {
   }
 
   // atm always recursive, but we should add some depth thing to it
-  list (folder = '/', { recursive = true } = {}) {
+  list (folder, opts) {
     if (typeof folder === 'object') return this.list(undefined, folder)
 
-    if (folder.endsWith('/')) folder = folder.slice(0, -1)
-    if (folder) folder = normalizePath(folder)
+    folder = std(folder || '/', true)
+
+    if (opts && opts.recursive === false) return shallowReadStream(this.db, folder, false)
 
-    if (recursive === false) return shallowReadStream(this.files, folder, false)
     // '0' is binary +1 of /
     return folder ? this.entries({ gt: folder + '/', lt: folder + '0' }) : this.entries()
```

Comment:
Something to consider, we are saying "if folder then use range, otherwise default" but we are not doing that in watch for example

Can't just assume to always use a range?

Reply:
Since we are normalizing all filenames

Or watch should do the same? (explicit condition to use range)